src/main/java/seedu/address/logic/commands/DeleteCommand.java

```diff
@@ -22,32 +24,112 @@
             + "Parameters: INDEX (must be a positive integer)\n"
             + "Example: " + COMMAND_WORD + " 1";
 
-    public static final String MESSAGE_DELETE_TASK_SUCCESS = "Deleted Task: %1$s";
+    private final List<Index> targetIndexes;
 
-    private final Index targetIndex;
-
-    public DeleteCommand(Index targetIndex) {
-        this.targetIndex = targetIndex;
+    public DeleteCommand(List<Index> targetIndexes) {
+        this.targetIndexes = MassOpsParser.sortInDesc(targetIndexes);
     }
 
     @Override
     public CommandResult execute(Model model) throws CommandException {
         requireNonNull(model);
         List<Task> lastShownList = model.getFilteredTaskList();
+        List<Task> deletedTasks = new ArrayList<>();
+        List<Index> deletedTasksIndexes = new ArrayList<>();
+        List<Index> outOfBoundsIndexes = new ArrayList<>();
 
-        if (targetIndex.getZeroBased() >= lastShownList.size()) {
-            throw new CommandException(Messages.MESSAGE_INVALID_TASK_DISPLAYED_INDEX);
+        for (int i = 0; i < targetIndexes.size(); i++) {
+            Index index = targetIndexes.get(i);
+            if (isValidIndex(index, lastShownList) == -1) {
+                outOfBoundsIndexes.add(index);
+            } else {
+                Task taskToDelete = lastShownList.get(index.getZeroBased());
+                model.deleteTask(taskToDelete);
+                deletedTasks.add(taskToDelete);
+                deletedTasksIndexes.add(index);
+            }
         }
-
-        Task taskToDelete = lastShownList.get(targetIndex.getZeroBased());
-        model.deleteTask(taskToDelete);
-        return new CommandResult(String.format(MESSAGE_DELETE_TASK_SUCCESS, taskToDelete));
+        return result(deletedTasks, deletedTasksIndexes, outOfBoundsIndexes);
     }
 
     @Override
     public boolean equals(Object other) {
         return other == this // short circuit if same object
                 || (other instanceof DeleteCommand // instanceof handles nulls
-                && targetIndex.equals(((DeleteCommand) other).targetIndex)); // state check
+                && targetIndexes.equals(((DeleteCommand) other).targetIndexes)); // state check
+    }
+
+    /**
+     * Converts the list of successfully deleted tasks into a string to be returned to the user
+     * @param deletedTasks
+     * @return
+     */
+    private String deletedTasksToString(List<Task> deletedTasks, List<Index> deletedTasksIndexes) {
```

Comment:
Should the `@return` description be filled as well? Also, according to the [coding standards](https://se-education.org/guides/conventions/java/intermediate.html#comments), I think there should be an empty line between the description and parameter section! This is noticed in a few other javadoc comments as well